deep_quoridor/src/agents/alphazero/nn_evaluator.py

```diff
@@ -36,20 +38,20 @@ def evaluate(self, game: Quoridor):
 
         self.network.eval()  # Disables dropout
 
+        # Prepare a tensor with True only on INVALID action positions so that we can set those values to
+        # a large negative number below, to discourage choosing them.
         valid_actions = game.get_valid_actions()
         valid_action_indices = [self.action_encoder.action_to_index(action) for action in valid_actions]
+        valid_indices_tensor = torch.tensor(valid_action_indices, device=self.device)
+        invalid_mask = torch.ones(self.action_encoder.num_actions, dtype=torch.bool, device=self.device)
+        invalid_mask[valid_indices_tensor] = False
 
         with torch.no_grad():
             input_array = self.game_to_input_array(game)
-            unmasked_policy_logits, value = self.network(torch.from_numpy(input_array).float().to(self.device))
-            unmasked_policy_logits = unmasked_policy_logits.cpu().numpy()
-
-            # Set logits of invalid actions to negative infinity to ensure they are never chosen
-            # TODO: Mask entirely in Torch to avoid going to Numpy and back
-            masked_logits = np.full_like(unmasked_policy_logits, -np.inf)
-            masked_logits[valid_action_indices] = unmasked_policy_logits[valid_action_indices]
-
-            policy_masked = F.softmax(torch.from_numpy(masked_logits), dim=-1).cpu().numpy()
+            policy_logits, value = self.network(torch.from_numpy(input_array).float().to(self.device))
+            assert torch.isfinite(policy_logits).all(), "Policy logits contains non-finite values"
```

Comment:
The lines after the assert will never do anything, right?

Reply:
(if we decide that getting -inf out of the network is expected behavior, then it will be nice to have those two lines)